### ISSUE TITLE
CBLAS detection does not work on Ubuntu 14.04

### ISSUE THREAD

**Issue by ghisvail** [arrayfire/arrayfire#1143]
Trying to generate the packaging for the older but still widely used Ubuntu 14.04 LTS, the CBLAS detection step fails, despite the headers and libs being [present](http://packages.ubuntu.com/trusty/amd64/libblas-dev/filelist) on the system.

The main difference between `libblas-dev` from Ubuntu 14.04 and current Debian is the lack of pkgconfig file, which means the CBLAS detection skips the pkgconfig part and goes through the alternative detection strategy. The latter somehow fails to recognize the Ubuntu packaged files.

**Comment by pavanky** [comment]
@ghisvail The alternative is looking for `libcblas` which is where most distros have the C wrapper around blas. It's going to be a pain to implement special cases for all the different distros. Ideally upstream should be dealing with, but we have added a bunch of special cases ourselves. 

That said, `atlas` an alternative to `cblas` works just fine on Ubuntu. Can't the package just depend on atlas ?

**Comment by ghisvail** [comment]
> Can't the package just depend on atlas ?

It could, but it would break support for the [alternatives mechanism](https://wiki.debian.org/DebianAlternatives).

> It's going to be a pain to implement special cases for all the different distros.

FYI, Ubuntu 14.04 is likely to stay for quite a while in research / production environments (still supported for 3+ years). But I can understand your reluctance to spend valuable dev time on this. I will maintain a patch for this particular issue downstream then.

**Comment by pavanky** [comment]
@ghisvail Can you check if the following patch works ?

```
diff --git a/CMakeModules/FindCBLAS.cmake b/CMakeModules/FindCBLAS.cmake
index d08b3c2..509eab5 100644
--- a/CMakeModules/FindCBLAS.cmake
+++ b/CMakeModules/FindCBLAS.cmake
@@ -36,7 +36,7 @@ IF(PC_CBLAS_FOUND)
     IF (NOT ${PC_LIB}_LIBRARY)
       message(FATAL_ERROR "Something is wrong in your pkg-config file - lib ${PC_LIB} not found in ${PC_CBLAS_LIBRARY_DIRS}")
     ENDIF (NOT ${PC_LIB}_LIBRARY)
-    LIST(APPEND CBLAS_LIBRARIES ${${PC_LIB}_LIBRARY}) 
+    LIST(APPEND CBLAS_LIBRARIES ${${PC_LIB}_LIBRARY})
   ENDFOREACH(PC_LIB)
 
   FIND_PACKAGE_HANDLE_STANDARD_ARGS(CBLAS DEFAULT_MSG CBLAS_LIBRARIES)
@@ -274,7 +274,7 @@ IF(NOT CBLAS_LIBRARIES)
     CBLAS
     cblas_dgemm
     ""
-    "cblas"
+    "cblas;blas"
     "cblas.h"
     TRUE,
     TRUE)
```

If it does, I think we can just add this.

**Comment by ghisvail** [comment]
I'll have a look and report back.

**Comment by pavanky** [comment]
I am working in a docker container to figure this out. The patch above doesn't work.

**Comment by pavanky** [comment]
@ghisvail found a patch that will work. We'll be pushing it to hotfixes-3.2.2

**Comment by pavanky** [comment]
@ghisvail #1146 should fix the issue you are seeing

**Comment by ghisvail** [comment]
I applied your fix to the packaging of `v3.2.1` and it worked for both Debian and Ubuntu.

**Comment by pavanky** [comment]
Fixed in hotfixes-3.2.2

**Comment by ghisvail** [comment]
Perfect. Thanks for taking care of this.
